File: 1.Data structure/6.PriorityQueue/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
		PriorityQueue<Integer> h = new PriorityQueue<>();
		int answer = 0;
		
		for(int i : scoville) {
			h.add(i);
		}
		
        while (h.size() >= 2 && h.peek() < K ) {
        	h.add(h.poll() + h.poll() * 2);
        	answer += 1;
        }
		return (h.peek() >= K ? answer: -1);
    }
}

File: 1.Data structure/6.PriorityQueue/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    answer = 0
    try:
        while scoville[0]<K:
            heapq.heappush(scoville, heapq.heappop(scoville)+heapq.heappop(scoville)*2)
            answer += 1
        return answer
    except:
        return -1